### PR TITLE
Only add prefix once in policy finder

### PR DESCRIPTION
**Pull request by mwq**
See issue (https://github.com/varvet/godmin/issues/87)

### CODE REVIEW

File: lib/godmin/authorization/policy_finder.rb [hunk 1]
```diff
@@ -17,10 +17,20 @@ def find(object)
...
-            "#{klass}Policy"
+            prefix = "#{Godmin.namespace.classify}::"
           end
+          policy_class = "#{klass}Policy"
+          add_prefix_once(prefix, policy_class)
+        end
+
+        private
+
+        def add_prefix_once(prefix, s)
```
Comment (Linuus) [inline lib/godmin/authorization/policy_finder.rb]: Name variable `s` a bit more informative :)

Comment (mwq) [inline lib/godmin/authorization/policy_finder.rb]: * is "random_duck_typed_object_that_can_be_interpolated" descriptive enough?
* interpolatable?

Comment (Linuus) [inline lib/godmin/authorization/policy_finder.rb]: No.

Comment (Linuus) [inline lib/godmin/authorization/policy_finder.rb]: `policy_class`?

Comment (mwq) [inline lib/godmin/authorization/policy_finder.rb]: Adding it to the string class:
```
class String
  def idempotent_prepend(prefix, str)
  end
end
```

**Comment by jensljungblad** [comment]
Perhaps something like this?

```ruby
def find(object)
  klass =
    if object.respond_to?(:model_name)
      object.model_name
    elsif object.class.respond_to?(:model_name)
      object.class.model_name
    elsif object.is_a?(Class)
      object
    elsif object.is_a?(Symbol)
      object.to_s.classify
    else
      object.class
    end

  "#{prefix(klass)}::#{klass}Policy"
end

private

def prefix(klass)
  return unless Godmin.namespace
  return if klass.start_with?(Godmin.namespace)
  Godmin.namespace.classify
end
```

File: lib/godmin/authorization/policy_finder.rb (hunk 1 above)
Comment (Linuus) [inline lib/godmin/authorization/policy_finder.rb]: No

**Pull request closed without merging**